crates/client/sync/src/fetch/fetchers.rs

```diff
@@ -143,12 +145,8 @@ pub async fn fetch_block_and_updates(
     Ok(converted)
 }
 
-async fn retry<F, Fut, T>(
-    mut f: F,
-    max_retries: u32,
-    base_delay: Duration,
-    ctx: &ServiceContext,
-) -> Result<T, SequencerError>
+// TODO: should we be checking for cancellation here? This might take a while
```

Comment:
i think this is cancellation safe and we can just drop the future here, it's just a fetch right?

Reply:
caller has to be aware of that

Reply:
I'll be looking into that and adding something like

```rust
tokio::select! {
    res = retry(...) => res,
    _ = ctx.cancelled() => return anyhow::Ok(),
}
```

Reply:
Actually this is not needed as we are already checking for cancellation when we are polling the block fetch stream. This means the block fetching might be cancelled, but this should be safe as we are not mutating the state of the node until those blocks are sent to the verify and apply task.